src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -101,4 +101,11 @@ void invalidateChannelIfSequencersMatch(
   int getSize() {
     return cache.size();
   }
+
+  /** Get the number of channels for a given table */
+  int getSizePerTable(String fullyQualifiedTableName) {
+    ConcurrentHashMap<String, SnowflakeStreamingIngestChannelInternal<T>> channelsMapPerTable =
+        cache.get(fullyQualifiedTableName);
```

Comment:
Is this thread safe? It's possible that the count changes between these two calls, right?